### PR TITLE
Fix swept_outputs issues

### PR DESCRIPTION
**Pull request by fireice-uk**
Bug found by mosu_forge.

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -8670,36 +8683,21 @@ uint64_t wallet2::import_key_images(const std::vector<std::pair<crypto::key_imag
 			spent += amount;
 		else
 			unspent += amount;
+
 		LOG_PRINT_L2("Transfer " << i << ": " << print_money(amount) << " (" << td.m_global_output_index << "): "
 								 << (td.m_spent ? "spent" : "unspent") << " (key image " << req.key_images[i] << ")");
 
-		if(i < daemon_resp.spent_status.size() && daemon_resp.spent_status[i] == COMMAND_RPC_IS_KEY_IMAGE_SPENT::SPENT_IN_BLOCKCHAIN)
+		if(daemon_resp.spent_status[i] == COMMAND_RPC_IS_KEY_IMAGE_SPENT::SPENT_IN_BLOCKCHAIN)
```
Comment (psychocrypt) [inline src/wallet/wallet2.cpp]: Why do you removed `i < daemon_resp.spent_status.size()`. What is if the daemon request will not have as much elements as `signed_key_images`, than we create an invalid memory access.

Comment (fireice-uk) [reply src/wallet/wallet2.cpp]: This is already tested for at https://github.com/ryo-currency/ryo-currency/blob/7a197b3232a38182508ac36346c1679e9bdb47a6/src/wallet/wallet2.cpp#L8651

File: src/wallet/wallet2.cpp [hunk 2]
```diff
@@ -8670,36 +8683,21 @@ uint64_t wallet2::import_key_images(const std::vector<std::pair<crypto::key_imag
 			spent += amount;
 		else
 			unspent += amount;
+
 		LOG_PRINT_L2("Transfer " << i << ": " << print_money(amount) << " (" << td.m_global_output_index << "): "
```
Comment (psychocrypt) [inline src/wallet/wallet2.cpp]: Before the loop it should be a exception if `req.key_images.size() > signed_key_images,size()`

File: src/wallet/wallet2.cpp (hunk 1 above)
Comment (mosu-forge) [reply src/wallet/wallet2.cpp]: I can confirm an exception is thrown if sizes are different

Comment (psychocrypt) [reply src/wallet/wallet2.cpp]: but it is only tested if `check_spent` is true.

https://github.com/ryo-currency/ryo-currency/blob/7a197b3232a38182508ac36346c1679e9bdb47a6/src/wallet/wallet2.cpp#L8643

File: src/wallet/wallet2.cpp [hunk 3]
```diff
@@ -8640,28 +8640,42 @@ uint64_t wallet2::import_key_images(const std::vector<std::pair<crypto::key_imag
 		m_transfers[n].m_key_image_partial = false;
 	}
 
-	if(check_spent)
+	if(!check_spent)
```
Comment (psychocrypt) [inline src/wallet/wallet2.cpp]: `spend` and `unspend` from https://github.com/ryo-currency/ryo-currency/pull/141/files?w=1#diff-9d580668dab930045943902048e2ac22R8661 should be set to zero before this early exit.

**Pull request merged**